src/main/java/metrics_influxdb/api/protocols/HttpInfluxdbProtocol.java

```diff
@@ -0,0 +1,63 @@
+package metrics_influxdb.api.protocols;
+
+public class HttpInfluxdbProtocol implements InfluxdbProtocol {
+    public final static String DEFAULT_HOST = "127.0.0.1";
+    public final static int DEFAULT_PORT = 8086;
+    public final static String DEFAULT_DATABASE = "metrics";
+    private final String user;
+    private final String password;
+    private final String host; 
+    private final int port;
+    private final boolean secured;
+    private final String database;
+    
+    public HttpInfluxdbProtocol(String host, int port, String user, String password, String db) {
+        super();
+        this.host = host;
+        this.port = port;
+        this.user = user;
+        this.password = password;
+        this.database = db;
+        this.secured = (user != null) && (password != null);
+    }
+    
+    public HttpInfluxdbProtocol(String host) {
+        this(host, DEFAULT_PORT);
+    }
+    
+    public HttpInfluxdbProtocol(String host, int port) {
+        this(host, port, null, null);
+    }
+    
+    public HttpInfluxdbProtocol(String host, int port, String database) {
+        this(host, port, null, null, database);
+    }
+    
+    public HttpInfluxdbProtocol() {
+        this(DEFAULT_HOST, DEFAULT_PORT);
+    }
+    
+    public HttpInfluxdbProtocol(String host, int port, String user, String password) {
+        this(host, port, user, password, DEFAULT_DATABASE);
+    }
+    
+    public String getUser() {
```

Comment:
I'm aware of this "advantages", and I applied it, but I prefer to have less code and simple API and to change the API if I change the behavior, and then notify user about this change (compilation break). http://dwayneb.free.fr/post/why_i_don_t_continue_to_use_getter_and_setter/